micro-deps/micro-deps-spring-test-config/src/test/groovy/com/ofg/infrastructure/stub/InteractionsSpec.groovy

```diff
@@ -0,0 +1,176 @@
+package com.ofg.infrastructure.stub
+
+import com.github.tomakehurst.wiremock.client.MappingBuilder
+import com.github.tomakehurst.wiremock.client.RequestPatternBuilder
+import com.github.tomakehurst.wiremock.client.ResponseDefinitionBuilder
+import com.github.tomakehurst.wiremock.client.UrlMatchingStrategy
+import com.github.tomakehurst.wiremock.client.VerificationException
+import com.github.tomakehurst.wiremock.client.WireMock
+import com.github.tomakehurst.wiremock.http.RequestMethod
+import com.google.common.base.Optional
+import com.ofg.infrastructure.discovery.ServiceConfigurationResolver
+import com.ofg.infrastructure.discovery.web.HttpMockServer
+import com.ofg.stub.StubRunning
+import groovyx.net.http.HTTPBuilder
+import spock.lang.AutoCleanup
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import static com.ofg.infrastructure.base.dsl.WireMockHttpRequestMapper.wireMockGet
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse
+
+class InteractionsSpec extends Specification {
+
+    static final String UNKOWN_COLLABORATOR = 'unknown collaborator'
+    static final String PING = 'ping collaborator'
+    static final String PONG = 'pong collaborator'
+    static final String UNKNOWN_PING_PATH = 'com/ofg/unknown/ping'
+    static final String KNOWN_PONG_PATH = 'com/ofg/pong'
+    static final int MOCK_PORT = 8993
+    static final String LOCALHOST = "localhost"
+    static final GString PONG_STUB_URL = "http://$LOCALHOST:$MOCK_PORT"
+    static final String PONG_ENDPOINT = '/pong'
+
+    @AutoCleanup('shutdownServer') HttpMockServer mockServer
+    @AutoCleanup('shutdown') WireMock wireMock
+    StubRunning stubRunning
+    ServiceConfigurationResolver configurationResolver
+
+    def setup() {
+        stubRunningWithPredefinedPongPath()
+        configurationResolverWithPingPongDependencies()
+
+        mockServer = new HttpMockServer(MOCK_PORT)
+        mockServer.start()
+
+        wireMock = new WireMock(LOCALHOST, mockServer.port())
+        wireMock.resetToDefaultMappings()
+    }
+
+    def 'should successfully verify interaction with stub'() {
+        given:
+            predefinedPongInteraction()
+            simulatedSinglePongInteraction()
+        when:
+            Interactions interactions = new Interactions(configurationResolver, stubRunning)
```

Comment:
Can we somehow add `Interactions` by default e.g. to instance of `com.ofg.infrastructure.base.MvcWiremockIntegrationSpec`?